src/lib/services/video-url-parser-service/video-url-parser-service.ts

```diff
@@ -35,7 +37,61 @@ export class VideoUrlParserService {
     return `https://player.vimeo.com/video/${vimeoVideoID}`;
   }
 
-  static getEmbedVideoURL(embedVideoURL: string): string {
+  static async tiktokParser(
+    backendApi: BackendApiService,
+    globalVars: GlobalVarsService,
+    url: string
+  ): Promise<string | boolean> {
+    let fullTikTokURL = url;
+    try {
+      const tiktokURL = new URL(url);
+      if (tiktokURL.hostname === "vm.tiktok.com") {
```

Comment:
If we're doing the validation serverside I'd skip this

Reply:
saves us a trip to the server, let's keep it.